commander/src/bootstrapping/commands/genesis-block/create.ts

```diff
@@ -0,0 +1,221 @@
+		address: Buffer.from(acc.address, 'hex'),
+		token: { balance: BigInt(tokenBalance) },
+	}));
+
+const prepareValidatorAccounts = (
+	data: {
+		username: string;
+		address: string;
+	}[],
+	tokenBalance: number,
+): Account[] =>
+	data.map(acc => ({
+		address: Buffer.from(acc.address, 'hex'),
+		token: { balance: BigInt(tokenBalance) },
+		dpos: {
+			delegate: {
+				username: acc.username,
+			},
+		},
+	}));
+
+export abstract class BaseGenesisBlockCommand extends Command {
+	static description = 'Creates genesis block file.';
+	static examples = [
+		'genesis-block:create --output mydir',
+		'genesis-block:create --output mydir --accounts 10',
+		'genesis-block:create --output mydir --accounts 10 --validators 103',
+		'genesis-block:create --output mydir --accounts 10 --validators 103 --token-distribution 500',
+	];
+
+	static flags = {
+		output: flagParser.string({
+			char: 'o',
+			description: 'Output folder path of the generated genesis block',
+			default: 'config',
+		}),
+		accounts: flagParser.integer({
+			char: 'a',
+			description: 'Number of non-validator accounts to generate',
+			default: 10,
+		}),
+		validators: flagParser.integer({
+			char: 'v',
+			description: 'Number of validator accounts to generate',
+			default: 103,
+		}),
+		'token-distribution': flagParser.integer({
+			char: 't',
+			description: 'Amount of tokens distributed to each account',
+			default: 10000000,
+		}),
+	};
+
+	async run(): Promise<void> {
+		const {
+			flags: { output, accounts, validators, 'token-distribution': tokenDistribution },
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		} = this.parse(BaseGenesisBlockCommand);
+
+		const accountList = new Array(accounts).fill(0).map(_x => createAccount());
+
+		const delegateList = new Array(validators).fill(0).map((_x, index) => ({
+			...{ username: `delegate_${index}` },
+			...createAccount(),
+			...{ password: createMnemonicPassphrase() },
+		}));
+
+		const onionSeed = cryptography.generateHashOnionSeed();
+		const onionCount = 10000; // These parameters can be configurable using relevant flags
+		const onionDistance = 1000;
+
+		const delegateForgingInfo = delegateList.map(del => ({
+			// ToDo: use a better password, user sourced using flag
+			encryptedPassphrase: cryptography.stringifyEncryptedPassphrase(
+				cryptography.encryptPassphraseWithPassword(del.passphrase, del.password),
+			),
+			hashOnion: {
+				count: onionCount,
+				distance: onionDistance,
+				hashes: cryptography
+					.hashOnion(onionSeed, onionCount, onionDistance)
+					.map(buf => buf.toString('hex')),
+			},
+			address: del.address,
+		}));
+
+		const validAccounts = prepareNormalAccounts(accountList, tokenDistribution);
+		const validDelegateAccounts = prepareValidatorAccounts(delegateList, tokenDistribution);
+		const app = this.getApplication({}, {});
+		const schema = app.getSchema();
+		const defaultAccount = app.getDefaultAccount();
+		if (!('token' in defaultAccount)) {
```

Comment:
prefer to use `getRegisteredModules` to check if the module is registered or not.
Also, ill create an issue to make it network agnostic